lib/grape/endpoint.rb

```diff
@@ -248,6 +248,8 @@ def run
 
         run_filters befores, :before
 
+        raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]) if env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]
```

Comment:
Sure.  It only is duplicated if it is present so I thought it was okay.  Do you have a preference between these two?

1:
```ruby
if (allowed_methods = env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED])
  raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => allowed_methods)
end
```

2:
```ruby
allowed_methods = env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]
raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => allowed_methods) if allowed_methods
```